app/api/scraper.py

```diff
@@ -28,15 +26,13 @@ def user_info_check(url: str, scraper: TwitterScraper = Depends()):
 
 @router.get("/detail", response_model=schemas.DetailResponse, name="user:get-detail")
 def user_detail_check(url: str, scraper: TwitterScraper = Depends()):
-    # Validate input url
-    if url[:20] == "https://twitter.com/":
-        username = url.split("/")[3]
-    elif url[:12] == "twitter.com/":
-        username = url.split("/")[1]
-    elif url != "" and "/" not in url:
+    if url == "":
+        raise HTTPException(400, "'url' argument is invalid!")
+
+    if "/" not in url:
         username = url
     else:
-        raise HTTPException(status_code=400, detail="'url' argument is invalid!")
+        username = parse_twitter_url(url)
```

Comment:
Cant we put all the input validation logic in the same module? Seeing half of it outside and half of it inside the `parse_twitter_url` is kinda itchy

Reply:
Because bare username is not URL?

Reply:
yea but :v, cant we just rename the function to `parse_twitter_username` :v?